src/mumble/AudioPreprocessor.cpp

```diff
@@ -0,0 +1,178 @@
+// Copyright 2024 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "AudioPreprocessor.h"
+
+#include <utility>
+
+#include <speex/speex_preprocess.h>
+
+AudioPreprocessor::AudioPreprocessor(AudioPreprocessor &&other) : m_handle(std::exchange(other.m_handle, nullptr)) {
+}
+
+AudioPreprocessor::~AudioPreprocessor() {
+	deinit();
+}
+
+AudioPreprocessor &AudioPreprocessor::operator=(AudioPreprocessor &&other) {
+	m_handle = std::exchange(other.m_handle, nullptr);
+	return *this;
+}
+
+bool AudioPreprocessor::init(const std::uint32_t sampleRate, const std::uint32_t quantum) {
+	deinit();
+
+	m_handle = speex_preprocess_state_init(static_cast< int >(quantum), static_cast< int >(sampleRate));
+	return m_handle != nullptr;
+}
+
+void AudioPreprocessor::deinit() {
+	if (m_handle) {
+		speex_preprocess_state_destroy(m_handle);
+	}
```

Comment:
This should probably also set`m_handle` to `nullptr` in case `deinit` gets called twice in a row.